workflows/differentialabundance.nf

```diff
@@ -360,172 +378,133 @@ workflow DIFFERENTIALABUNDANCE {
+                meta,
+                matrix,
+                tools_diff.method,
+                tools_diff.fc_threshold,
+                tools_diff.stat_threshold
+            ]
+        }
 
-    if (params.study_type == 'affy_array' ||
-        params.study_type == 'geo_soft_file' ||
-        params.study_type == 'maxquant' ||
-        (params.study_type == 'rnaseq' && params.differential_use_limma)
-        ) {
+    // Run differential analysis
 
-        LIMMA_DIFFERENTIAL (
-            ch_contrasts,
-            ch_samples_and_matrix
-        )
-        ch_differential = LIMMA_DIFFERENTIAL.out.results
-        ch_model = LIMMA_DIFFERENTIAL.out.model
+    ABUNDANCE_DIFFERENTIAL_FILTER(
+        ch_differential_input,
+        VALIDATOR.out.sample_meta,
+        ch_transcript_lengths,
+        ch_control_features,
+        ch_contrasts
+    )
 
-        ch_versions = ch_versions
-            .mix(LIMMA_DIFFERENTIAL.out.versions)
+    // collect differential results
 
-        if (params.study_type == 'rnaseq') {
-            ch_norm = LIMMA_DIFFERENTIAL.out.normalised_counts.first()
-        }
+    ch_differential_results = ABUNDANCE_DIFFERENTIAL_FILTER.out.results_genewise
+    ch_differential_results_filtered = ABUNDANCE_DIFFERENTIAL_FILTER.out.results_genewise_filtered
+    ch_differential_model = ABUNDANCE_DIFFERENTIAL_FILTER.out.model
+    ch_differential_norm = ABUNDANCE_DIFFERENTIAL_FILTER.out.normalised_matrix
+    ch_differential_varstab = ABUNDANCE_DIFFERENTIAL_FILTER.out.variance_stabilised_matrix
 
-        ch_processed_matrices = ch_norm
-            .map{ it.tail() }
-            .first()
+    ch_versions = ch_versions
+        .mix(ABUNDANCE_DIFFERENTIAL_FILTER.out.versions)
 
+    // when the study type is rnaseq, we use the normalised matrix from the differential subworkflow
+    if (params.study_type == 'rnaseq') {
+        ch_norm = ch_differential_norm
     } else {
-        DESEQ2_NORM (
-            ch_contrasts.first(),
-            ch_samples_and_matrix,
-            ch_control_features,
-            ch_transcript_lengths
-        )
-
-        // Run the DESeq differential module, which doesn't take the feature
-        // annotations
+        // when the study type is not rnaseq, the normalised matrix comes directly from VALIDATOR.out
+        // we need to update the meta so that it can match with tools_norm
+        ch_norm = ch_norm
+            .map { meta, norm ->
+                def meta_new = meta + [method_differential: 'validator']
+                [meta_new, norm]
+            }
+    }
 
-        DESEQ2_DIFFERENTIAL (
-            ch_contrasts,
-            ch_samples_and_matrix,
-            ch_control_features,
-            ch_transcript_lengths
+    // prepare channel with normalized matrix, and variance stabilized matrices when available
+    ch_processed_matrices = ch_norm
+        .filter { meta, norm -> meta.method_differential == 'deseq2' } // only deseq2 produce variance stabilized matrices
```

Comment:
Sorry, think I've missed something. In the current logic ch_processed matrices is populated as the comment above- i.e. it contains the vs matrix where available. 

What is the reason for filtering to only the deseq2 results here? Adjust the comment accordingly